CONTRIBUTING.md

```diff
@@ -269,3 +269,40 @@ Another option is to run one or more tests directly:
 
 This will still install the task and run `pre-apply-task-hook.sh` if present, but it will then
 run only the specified test pipeline.
+
+### Openshift CI Prow Tests
+
+Each PR runs the [release-pipelines e2e test suite](https://github.com/redhat-appstudio/e2e-tests/tree/main/tests/release/pipelines).
+This is to ensure that a change does not break any pipeline functionality, as the tekton unit tests only test the tasks.
+This shows up as the `ci/prow/release-pipelines-e2e-suite` check on the PR.
+
+#### Pairing with E2E Change
+
+Sometimes, a change to a pipeline definition will require a change to the e2e test in order to pass. To do this, the changes should be paired.
+
+Process:
+  This is described in depth [here](https://github.com/redhat-appstudio/e2e-tests/blob/main/docs/Installation.md#rhtap-in-openshift-ci-and-branch-pairing).
+  The short explanation for it is that you should open a PR that makes the necessary changes to the test suite in
+  [the e2e repo](https://github.com/redhat-appstudio/e2e-tests) using the same branch name as your PR to this repository. This will pair them. Once the
+  PR to this repo is merged, the e2e-tests PR should also be merged so that future PRs to this repo will pass (as they will now be dependent on the e2e change).
+
+#### Pairing with release-service Change
+
+Another possible pairing scenario is a change to a pipeline definition that requires changes to the operator itself, stored in
+[the release-service repo](https://github.com/redhat-appstudio/release-service). For example, maybe a parameter is passed by the operator
+to the pipeline, and the name of the parameter is changing. This will require the release-service PR and release-service-catalog PR to be paired
+in order for the e2e suite to pass.
+Note: This can be used in conjunction with the e2e pairing described in the previous section. That is to say, you can also pair an e2e-tests
+PR with these two PRs.
+
+Process:
+* Open a PR to the [release-service](https://github.com/redhat-appstudio/release-service). The catalog PR will run using the image that is built in the
+`ci/prow/images` check on the release-service PR. So, it is important to open the release-service PR first and wait for that check to succeed.
```

Comment:
> wait for that check to succeed

But how will it succeed if you need to pair two breaking changes together? Consider the example where you change a pipeline parameter name. When you open the release-service PR to change the parameter name that the operator provides when creating the pipeline, the e2e run might fail, because it uses the catalog before the required change, no?

Reply:
If run, yeah. The way I did it in my test is to open the release-service PR as a draft so the e2e stuff doesn't run, then do `/test images` so just that one check runs and the image is built. But yeah, the e2e suite on the release-service PR wouldn't pass at this point

Reply:
Hmm, so maybe it's worth adding those bits here? That way one e2e run is saved and also the team is not spammed about reviewing the release-service PR. What's missing is also what to do once the catalog PR is merged here. At that point you're in a broken state, so you should mark the release-service PR as Ready and the e2e run will start and it should pass and then it should be merged ASAP, right?

Reply:
I updated it, can you take a look and tell me if it reads okay?

Reply:
I like the updated version, thanks!